cmake/external/cares.cmake

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
```

Comment:
Are `cares` and `grpc` needed in all kinds of builds of Paddle? I just suggest to add an `if` statement here to return early if they are not needed, or at least add the following statements to avoid the failing of building for mobile:
```
IF(MOBILE_INFERENCE)
    return()
ENDIF()
```

Reply:
Sure. Will add. Thanks for reminding.

Reply:
Done.